### PR TITLE
[Improve]Improve import speed by compressing data at low internet speeds

### PR DESCRIPTION
**Pull request by fluozhiye**
# Proposed changes

Improve import speed by compressing data at low internet speeds。The lower the bandwidth, the better the effect。

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: spark-doris-connector/src/main/java/org/apache/doris/spark/load/DorisStreamLoad.java [hunk 1]
```diff
@@ -222,13 +226,34 @@ public long load(Iterator<InternalRow> rows, StructType schema)
...
-                    .format(dataFormat)
-                    .sep(FIELD_DELIMITER)
-                    .delim(LINE_DELIMITER)
-                    .schema(schema)
-                    .addDoubleQuotes(addDoubleQuotes).build(), streamingPassthrough);
-            httpPut.setEntity(new InputStreamEntity(recodeBatchInputStream));
+
+            if(StringUtils.isNotEmpty(compressType)){
+                if("gz".equals(compressType.toLowerCase())){
+                    if(dataFormat.equals(DataFormat.CSV)){
```
Comment (JNSimba) [inline spark-doris-connector/src/main/java/org/apache/doris/spark/load/DorisStreamLoad.java:232]: Is it possible to judge directly this way?  `if("gz".equals(compressType.toLowerCase()) && dataFormat.equals(DataFormat.CSV) )`

File: spark-doris-connector/src/main/java/org/apache/doris/spark/load/DorisStreamLoad.java [hunk 2]
```diff
@@ -505,4 +530,19 @@ private void handleStreamPassThrough() {
...
+    private byte[] compressByGZ(String content) throws IOException{
+        byte[] compressedData;
+        try(ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            GZIPOutputStream gzipOutputStream = new GZIPOutputStream(baos);
+        ){
+            gzipOutputStream.write(content.getBytes("UTF-8"));
+            gzipOutputStream.finish();
+            compressedData = baos.toByteArray();
+        }
+        return compressedData;
```
Comment (JNSimba) [inline spark-doris-connector/src/main/java/org/apache/doris/spark/load/DorisStreamLoad.java:545]: This method recommends adding a unit test

**Comment by JNSimba** [comment]
Thanks @fluozhiye , I left some comments, PTAL

File: spark-doris-connector/src/main/java/org/apache/doris/spark/load/DorisStreamLoad.java (hunk 1 above)
Comment (fluozhiye) [reply spark-doris-connector/src/main/java/org/apache/doris/spark/load/DorisStreamLoad.java]: No problem。I have made an update。

File: spark-doris-connector/src/main/java/org/apache/doris/spark/load/DorisStreamLoad.java (hunk 2 above)
Comment (fluozhiye) [reply spark-doris-connector/src/main/java/org/apache/doris/spark/load/DorisStreamLoad.java]: have submitted a unit test for the method

**Review by JNSimba — APPROVED** [review]
LGTM

File: spark-doris-connector/src/test/java/org/apache/doris/spark/load/TestDorisStreamLoad.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
+package org.apache.doris.spark.load;
```
Comment (JNSimba) [inline spark-doris-connector/src/test/java/org/apache/doris/spark/load/TestDorisStreamLoad.java:1]: please add license header, thanks

Comment (fluozhiye) [reply spark-doris-connector/src/test/java/org/apache/doris/spark/load/TestDorisStreamLoad.java:18]: added， thanks ！

**Review by JNSimba — APPROVED** [review]
LGTM

**Pull request merged**